From ef70ba7ba6b1bad9d873dfa48458fe16269fe48b Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Thu, 12 Feb 2015 09:06:41 +0100
Subject: [PATCH] banner

---
 src/usr/autoboot.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/usr/autoboot.c b/src/usr/autoboot.c
index 106e0f879..209fb868d 100644
--- a/src/usr/autoboot.c
+++ b/src/usr/autoboot.c
@@ -70,6 +70,7 @@ static int ( * is_autoboot_device ) ( struct net_device *netdev );
 
 #define NORMAL	"\033[0m"
 #define BOLD	"\033[1m"
+#define BLUE	"\033[34m"
 #define CYAN	"\033[36m"
 
 /** The "scriptlet" setting */
@@ -582,9 +583,13 @@ int ipxe ( struct net_device *netdev ) {
 	 * do so.
 	 *
 	 */
-	printf ( NORMAL "\n\n" PRODUCT_NAME "\n" BOLD PRODUCT_SHORT_NAME " %s"
+	printf ( NORMAL "\n\n" PRODUCT_NAME "\n" BOLD PRODUCT_SHORT_NAME
 		 NORMAL " -- " PRODUCT_TAG_LINE " -- "
-		 CYAN PRODUCT_URI NORMAL "\nFeatures:", product_version );
+		 CYAN PRODUCT_URI NORMAL "\n"
+		 BOLD "Arch" BLUE "Linux" NORMAL " -- make it simple and lightweight -- "
+		 BOLD BLUE "http://archlinux.org" NORMAL "\n"
+		 "Version %s compiled on " __DATE__ ", " __TIME__ "\n"
+		 "Features:", product_version );
 	for_each_table_entry ( feature, FEATURES )
 		printf ( " %s", feature->name );
 	printf ( "\n" );
