diff --git a/drivers/net/wireless/ath/ath10k/mac.c b/drivers/net/wireless/ath/ath10k/mac.c
index a6d21856b7e7..9417c469136e 100644
--- a/drivers/net/wireless/ath/ath10k/mac.c
+++ b/drivers/net/wireless/ath/ath10k/mac.c
@@ -8332,7 +8332,7 @@ static const struct ieee80211_ops ath10k_ops = {
 	.hw_value		= (_channel), \
 	.center_freq		= (_freq), \
 	.flags			= (_flags), \
-	.max_antenna_gain	= 0, \
+	.max_antenna_gain	= 12, \
 	.max_power		= 30, \
 }
 
@@ -8341,7 +8341,7 @@ static const struct ieee80211_ops ath10k_ops = {
 	.hw_value		= (_channel), \
 	.center_freq		= (_freq), \
 	.flags			= (_flags), \
-	.max_antenna_gain	= 0, \
+	.max_antenna_gain	= 12, \
 	.max_power		= 30, \
 }
 
